packages/components/src/Combobox/Combobox.tsx

```diff
@@ -74,7 +78,8 @@ export function Combobox(props: ComboboxProps): JSX.Element {
           wrapperRef={wrapperRef}
           open={open}
           setOpen={setOpen}
-          options={options}
+          options={filteredOptions}
```

Comment:
decided that the sub components really don't need to know or care that the options they are receiving are filtered or not

just render them if you get them

opted to add one more prop of `hadInitialOptions` since there is a different flow we want to happen if you're searching and got nothing as a result of filtering VS if you never had anything to begin with